contribs/gmf/src/services/themesservice.js

```diff
@@ -103,24 +120,51 @@ gmf.Themes.findTheme_ = function(themes, themeName) {
  * @return {angular.$q.Promise} Promise.
  */
 gmf.Themes.prototype.getBgLayers = function() {
+
   goog.asserts.assert(!goog.isNull(this.promise_));
   return this.promise_.then(goog.bind(
       /**
        * @param {gmf.ThemesResponse} data The "themes" web service response.
-       * @return {Array.<Object>} Array of background layer objects.
+       * @return {angular.$q.Promise}
        */
       function(data) {
-        var bgLayers = data['background_layers'].map(goog.bind(function(item) {
+        var configs = data['background_layers'];
+        this.backgroundLayerConfigs_ = configs;
+        var promises = configs.map(goog.bind(function(item) {
+          return this.$http_.get(item.url, {
+            cache: false
```

Comment:
> Does that make sense ?

Yes but, 
- cache is reset on page reload
- if you have 10 bg layers from same service, we don't want 10 request on init

You are right because the theme tree is loaded if you connect and disconnect, at this time, we want a fresh response.
Leave `false`, i will do some work on top of it afterwards too optimizes regarding the cases.